certbot/certbot/_internal/cli/subparsers.py

```diff
@@ -44,15 +44,15 @@ def _create_subparsers(helpful: "helpful.HelpfulArgumentParser") -> None:
                 "--delete-after-revoke", action="store_true",
                 default=flag_default("delete_after_revoke"),
                 help="Delete certificates after revoking them, along with all previous and later "
-                "versions of those certificates.")
+                "versions of those certificates. (default: False)")
     helpful.add("revoke",
                 "--no-delete-after-revoke", action="store_false",
                 dest="delete_after_revoke",
                 default=flag_default("delete_after_revoke"),
                 help="Do not delete certificates after revoking them. This "
                      "option should be used with caution because the 'renew' "
                      "subcommand will attempt to renew undeleted revoked "
-                     "certificates.")
+                     "certificates. (default: True)")
```

Comment:
```suggestion
                     "certificates. (default: False)")
```
i think this should be false. setting `no-delete-after-revoke = False` in cli.ini doesn't seem to change the default behavior for me and it kinda sorta makes sense depending on how you think about it. is the flag `--no-delete-after-revoke` set by default? false. do you want to set this flag in cli.ini? set it to true

this is what we seem to do for the other `store_false` flags like this too

Reply:
Oooh I hate this. 
```
constants.py True --> --no- + store_false --> default = False
constants.py True --> --  [we don't have any of these]
constants.py False --> -- + store_true --> default = False
constants.py False --> --no- + store_false --> default = False
```
I feel like this only makes sense if you know that we're doing `store_false` internally, because setting a `store_false` flag to `False` means "don't store it False" which is not the same as "store it True".

Setting the default to True is an accurate description of the behavior of the program -- the behavior described here is true by default! But I agree that it's also misleading in terms of what setting the flag to the opposite behavior would do. The original issue happened to make sense to request because it *wasn't* a `--no-` flag, but (preposterously) setting the parameter of a no-parameter flag isn't actually always guaranteed to work.

As for what we currently do, I feel like that's more a "we just used `flag_default`" in a couple cases, but more often we either suppressed the `--no-` option or manually specified what the flag does (eg. "Not reusing private keys is the default behavior of Certbot. This option may be used to unset --reuse-key on an existing certificate," "(default: redirect enabled for install and run, disabled for enhance)"). I think that's the actual answer -- just use our words, or suppress it.

And in the spirit of "just use our words," it sure is a little weird that we say "this should be used with caution"...but that's the default behavior.

Reply:
Anyway I just realized I didn't say it, but I think based on this logic the correct change here would be to go in and just specify more explicitly what's going on, like in the other examples. And similarly for `--no-eff-email`.

Reply:
Ugh doing it the way I want probably means adding an ignore case to the `CustomHelpFormatter`, which is probably correct but means I have to audit uses of `default` in help text.

Reply:
Nvm too many alternate uses of the word default

Reply:
lol, the actual default is Ask here as well.

Reply:
that explains the weird "default-but-cautioned behavior" thing; since the default is None, not False, that's not actually the default at all.

Reply:
brb checking to see if any of these others are also actually ask

Reply:
ok, audited.